ipa-core/src/protocol/dp/mod.rs

```diff
@@ -127,82 +294,62 @@ fn c_p(success_prob: f64) -> f64 {
             + 2.0 * (1.0 - success_prob).powi(2))
 }
 /// equation (16)
-#[allow(dead_code)]
 fn d_p(success_prob: f64) -> f64 {
     (4.0 / 3.0) * (success_prob.powi(2) + (1.0 - success_prob).powi(2))
 }
 /// equation (7)
-#[allow(clippy::too_many_arguments)]
-#[allow(dead_code)]
-fn epsilon_constraint(
-    num_bernoulli: u32,
-    success_prob: f64,
-    delta: f64,
-    quantization_scale: f64,
-    dimensions: f64,
-    ell_1_sensitivity: f64,
-    ell_2_sensitivity: f64,
-    ell_infty_sensitivity: f64,
-) -> f64 {
+fn epsilon_constraint(num_bernoulli: u32, noise_params: &NoiseParams) -> f64 {
     let num_bernoulli_f64 = f64::from(num_bernoulli);
-    let first_term_num = ell_2_sensitivity * (2.0 * (1.25 / delta).ln()).sqrt();
-    let first_term_den =
-        quantization_scale * (num_bernoulli_f64 * success_prob * (1.0 - success_prob)).sqrt();
-    let second_term_num = ell_2_sensitivity * c_p(success_prob) * ((10.0 / delta).ln()).sqrt()
-        + ell_1_sensitivity * b_p(success_prob);
-    let second_term_den = quantization_scale
+    let first_term_num =
+        noise_params.ell_2_sensitivity * (2.0 * (1.25 / noise_params.delta).ln()).sqrt();
+    let first_term_den = noise_params.quantization_scale
```

Comment:
I think this code is still very very dense and challenging to follow

Reply:
yeah, you're right.  But Thm1 of the [paper](https://arxiv.org/pdf/1805.10559) that we implement here is very dense too -- I think keeping it as close to the paper is what lets us check most easily that it is computing the correct thing. 

![image](https://github.com/user-attachments/assets/1bda4b41-2025-45d9-bcc7-b6de9a7fb87e)